### PR TITLE
Bug fix for controlnetpipeline check_image

### PR DESCRIPTION
**Pull request by Fantast616**
Bug fix for controlnetpipeline check_image when using multicontrolnet and prompt list

Fixes # (issue)
[https://github.com/huggingface/diffusers/issues/7102](https://github.com/huggingface/diffusers/issues/7102)

@sayakpaul

### CODE REVIEW

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Comment by yiyixuxu** [comment]
can we add a test here? for multi-controlnet and prompt list ?  https://github.com/huggingface/diffusers/blob/ad310af0d65d5a008401ebde806bed413156cf82/tests/pipelines/controlnet/test_controlnet.py#L280

**Comment by Fantast616** [comment]
> can we add a test here? for multi-controlnet and prompt list ?
> 
> https://github.com/huggingface/diffusers/blob/ad310af0d65d5a008401ebde806bed413156cf82/tests/pipelines/controlnet/test_controlnet.py#L280

OK Done, It already had an multiple prompt test before: test_inference_multiple_prompt_input(), but its' prompt size is equals to controlnet number, so it didn't find the problem. I modify the input prompt size to 4 now.

**Comment by sayakpaul** [comment]
@yiyixuxu could you give this a look again?

**Comment by yiyixuxu** [comment]
@Fantast616 
test still fails - can you look into that?

File: tests/pipelines/controlnet/test_controlnet.py [hunk 1]
```diff
@@ -475,12 +475,12 @@ def test_inference_multiple_prompt_input(self):
         sd_pipe.set_progress_bar_config(disable=None)
 
         inputs = self.get_dummy_inputs(device)
-        inputs["prompt"] = [inputs["prompt"], inputs["prompt"]]
-        inputs["image"] = [inputs["image"], inputs["image"]]
+        inputs["prompt"] = [inputs["prompt"], inputs["prompt"], inputs["prompt"], inputs["prompt"]]
```
Comment (yiyixuxu) [inline tests/pipelines/controlnet/test_controlnet.py:478]: instead of update this test, can you add a new one?
I think we also accept inputs like this

Comment (yiyixuxu) [reply tests/pipelines/controlnet/test_controlnet.py:478]: it can be in the same test, just add a few lines to test new input format

Comment (Fantast616) [reply tests/pipelines/controlnet/test_controlnet.py:478]: OK I will add tonight later

**Comment by yiyixuxu** [comment]
can you fix the quality test?

**Comment by Fantast616** [comment]
> can you fix the quality test?

Sorry for having other things recently, which may not be modified in time

**Comment by yiyixuxu** [comment]
gentle pin @Fantast616

**Pull request merged**